components/elm/src/biogeochem/PhosphorusStateUpdate3Mod.F90

```diff
@@ -181,7 +181,7 @@ subroutine PhosphorusStateUpdate3(bounds,num_soilc, filter_soilc, num_soilp, fil
                             col_pf%sminp_to_plant_vr(c,j)*dt - col_pf%labilep_to_secondp_vr(c,j)*dt - &
                             col_pf%sminp_leached_vr(c,j)*dt ))
 
-                 if (temp_solutionp(c,j) < 0.0_r8) then
+                 if (temp_solutionp(c,j) < 0.0_r8 .and. (col_pf%labilep_to_secondp_vr(c,j)+col_pf%sminp_leached_vr(c,j) .ne. 0._r8) ) then
```

Comment:
@qzhu-lbl if `col_pf%labilep_to_secondp_vr(c,j)+col_pf%sminp_leached_vr(c,j) = 0._r8`, the `col_pf%labilep_to_secondp_vr(c,j)` would be `nan`.  Is this code modification correct or would you recommend some other change?

Reply:
@bishtgautam the master branch recently got a fix for this issue -- that's the one conflict for this PR it seems. So we could just opt for the change in master during the merge.